backend/azure/dtmb/dtmb.go

```diff
@@ -0,0 +1,531 @@
+package dtmb
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/microsoft/durabletask-go/api"
+	"github.com/microsoft/durabletask-go/backend"
+	"github.com/microsoft/durabletask-go/internal/protos"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	dtmbprotos "github.com/microsoft/durabletask-go/backend/azure/dtmb/internal/backend/v1"
+	"github.com/microsoft/durabletask-go/backend/azure/dtmb/internal/utils"
+)
+
+const (
+	// DefaultEndpoint is the default endpoint for the DTMB service.
+	defaultEndpoint = "localhost:50051"
+)
+
+type dtmb struct {
+	logger                    backend.Logger
+	endpoint                  string
+	options                   *DTMBOptions
+	orchestrationQueue        utils.SyncQueue[dtmbprotos.ExecuteOrchestrationMessage]
+	activityQueue             utils.SyncQueue[dtmbprotos.ExecuteActivityMessage]
+	orchestrationHistoryCache utils.OrchestrationHistoryCache
+	connectWorkerClientStream chan *dtmbprotos.ConnectWorkerClientMessage
+	clientClient              dtmbprotos.TaskHubClientClient
+	workerClient              dtmbprotos.TaskHubWorkerClient
```

Comment:
The reason it's split into two is because one could have a client-only app, so the methods are separate.

However, @berndverst you don't need 2 objects. You can create something like:

```go
type TaskHubClient struct {
    dtmbprotos.TaskHubClientClient
    dtmbprotos.TaskHubWorkerClient
}
```

Then use embedding to add both "clients" to this, and you have a single object.

It's actually meant to be used this way when you have an app that needs both.